Incompatibility with networkx 2.0

Conda recently upgraded to NetworkX, which is triggering this traceback:

```
>       return load_model('timeseries2.json', solver=solver)

tests/test_recorders.py:1061: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

tests/helpers.py:18: in load_model

    model.check()

pywr/_model.pyx:139: in pywr._model.Model.check

    ???

pywr/_model.pyx:707: in pywr._model.Model.find_orphaned_parameters

    ???

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???

E   TypeError: object of type 'dict_keyiterator' has no len()

pywr/_component.pyx:44: TypeError
```

This is the cause of the failures in #527 and the current master build.

--- a/pywr/_component.pyx
+++ b/pywr/_component.pyx
@@ -37,7 +37,7 @@ class GraphInterface:
         if self is self.obj.children:
             return [n for n in self.graph.predecessors(self.obj) if n != ROOT_NODE]
         else:
-            return self.graph.successors(self.obj)
+            return [x for x in self.graph.successors(self.obj)]
 
     def __len__(self):
         """Returns the number of nodes in the model"""

--- a/pywr/nodes.py
+++ b/pywr/nodes.py
@@ -77,7 +77,7 @@ class Connectable(object):
             self._disconnect(node, slot_name=slot_name, all_slots=all_slots)
         else:
             neighbors = self.model.graph.neighbors(self)
-            for neighbor in neighbors:
+            for neighbor in [neighbor for neighbor in neighbors]:
                 self._disconnect(neighbor, slot_name=slot_name, all_slots=all_slots)
 
     def _disconnect(self, node, slot_name=None, all_slots=True):

